libraries/lib-xml/XMLAttributeValueView.h

```diff
@@ -0,0 +1,174 @@
+ * If the type is not large enough to store the value, Get fails.
+ *
+ * Floating point values can be read from the view, if it is initialized using floating point, integer or
+ * a compatible string value. Conversion from Float to Double is forbidden.
+ *
+ * String values can be read only if the view was created using a string_view.
+ * However there are ToString() and ToWString() methods that allow conversion from any type to string.
+ * Null type is represented by an empty string.
+ */
+class XML_API XMLAttributeValueView final
+{
+public:
+   //! Type of the value represented by the XMLAttributeValueView.
+   enum class Type
+   {
+      Null, //!< The value is not initialized.
+      SignedInteger, //!< The value is a signed integer.
+      UnsignedInteger, //!< The value is an unsigned integer.
+      Float, //!< The value is a single precision floating point value.
+      Double, //!< The value is a double precision floating point value.
+      StringView //!< The value is a string_view.
+   };
+
+   //! Construct an unitialized view of type Null
+   XMLAttributeValueView() = default;
+
+   XMLAttributeValueView(const XMLAttributeValueView&) = default;
+   XMLAttributeValueView(XMLAttributeValueView&&) = default;
+
+   XMLAttributeValueView& operator = (const XMLAttributeValueView&) = default;
+   XMLAttributeValueView& operator = (XMLAttributeValueView&&) = default;
+
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(bool value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(short value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned short value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(int value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned int value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(long value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned long value) noexcept;
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(long long value) noexcept;
+   //! Construct a view of type UnsignedInteger from the value.
+   explicit XMLAttributeValueView(unsigned long long value) noexcept;
+   //! Construct a view of type Float from the value.
+   explicit XMLAttributeValueView(float value) noexcept;
+   //! Construct a view of type Double from the value.
+   explicit XMLAttributeValueView(double value) noexcept;
+   //! Construct a view of type StringView from the value.
+   explicit XMLAttributeValueView(const std::string_view& value) noexcept;
```

Comment:
Isn't string_view cheap to pass by value?

Reply:
It is